README.md

```diff
@@ -2,24 +2,56 @@
 
 Proposal for syntax to import ES modules with attributes
 
+Proposed by [littledan](https://github.com/littledan) and [xtuc](https://github.com/xtuc).
+
 ## Motivation
 
 https://github.com/w3c/webcomponents/issues/839
 
-### When would the checks happen
+"The author is the only true arbiter of the parse goal of content".
+
+### Semantics
+
+On the web, before any code executes this semantics apply before any code executes.
+
+#### `type`
+
+Check that the mimetype matches what type was specified in the attributes.
 
-On the web, before any code executes, check that the mimetype matches what type was provided explicitly.
+#### `integrity`
+
+Check the hash of the resource.
 
 ## Integrations
 
 ### import statements
 
+The ImportDeclaration would allow any arbitrary attributes after the `with` keyword.
+
+A `type` key should be mandatory to load a module that isn't JavaScript (json, css, html, ...).
+
 ```js
 import json from "./foo.json" with type: "json";
 ```
 
+Optionally the developers can specify an SRI hash which semantically works like an assertions:
+
+```js
+import json from "./foo.json" with
+    type: "json",
+    integrity: "sha384-some-hash";
+```
+
+#### Builtin modules
```

Comment:
I don't really think we should include built-in modules here. Let's let that proceed separately first.